samza-core/src/main/java/org/apache/samza/startpoint/StartpointSerde.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.startpoint;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import org.apache.samza.SamzaException;
+import org.apache.samza.serializers.Serde;
+import org.codehaus.jackson.map.ObjectMapper;
+
+
+class StartpointSerde implements Serde<Startpoint> {
+  private static final String STARTPOINT_CLASS = "startpointClass";
+  private static final String STARTPOINT_OBJ = "startpointObj";
+
+  private final ObjectMapper mapper = new ObjectMapper();
+  private final HashMap<String, Class<? extends Startpoint>> registry = new HashMap<>();
+
+  void register(Class<? extends Startpoint> clazz) {
+    registry.put(clazz.getCanonicalName(), clazz);
+  }
+
+  @Override
+  public Startpoint fromBytes(byte[] bytes) {
+    try {
+      LinkedHashMap<String, String> deserialized = mapper.readValue(bytes, LinkedHashMap.class);
+      return mapper.readValue(deserialized.get(STARTPOINT_OBJ), registry.get(deserialized.get(STARTPOINT_CLASS)));
```

Comment:
You could use Class.forName to avoid needing a "registry". The disadvantage of using Class.forName is that you have to make sure to handle unexpected (or even malicious, if possible) class names in the store. Up to you.

Reply:
I think you would need to cast the resulting Class to a Class<? extends Startpoint>. But I think there is a method in Class to check if it extends Startpoint and then do the cast.